### PR TITLE
Add documentation for magazines [CR]

### PR DESCRIPTION
**Pull request by mugling**
In particular discussion/review of the balance guidelines set out in `GAME_BALANCE.md`

### CODE REVIEW

File: doc/GAME_BALANCE.md [hunk 1]
```diff
@@ -141,6 +141,47 @@ At high skill, bashing damage is generally the strongest, but still suffers from
...
+### Reliability
+Should be specified first considering the below and then scaled against any equivalent magazines. For example if an extended version of a magazine exists place it one rank below the standard capacity version. Damaged guns or magazines will further adversely affect reliability.
+
+10 - **Perfectly reliable**. Factory specification or milspec only. Never extended magazines. Very rare.
+
+9 - **Reliable**. Failures only in burst fire. Factory or milspec magazines only. Never extended magazines. Uncommon.
+
+8 - **Dependable**. Failures infrequently in any fire mode. Highest reliability possible for extended magazines and those crafted using gunsmithing tools. Most common.
+
+7 - **Serviceable**. Fail infrequently in semi-automatic, more frequently in burst. Includes many extended and aftermarket gunsmithing toosl. Common.
```
Comment (illi-kun) [inline doc/GAME_BALANCE.md]: typo: toosl -> tools

**Comment by chaosvolt** [comment]
I still don't really agree with having video-gamey weight ratios for magazines. It's not going to have that substantial an effect on gameplay, we're talking in terms of grams-per-capacity when a stockpile of ammo and the gun itself will end up being massively heavier overall.

It DOES however have a realism effect, and if it has no real gameplay purpose then there's no reason not to use real-world weights for the magazines whenever possible.

Plus it's inevitably going to draw complaints from people who are even bigger gun nerds than I am. XP

**Comment by mugling** [comment]
> I still don't really agree with having video-gamey weight ratios for magazines

At least they will be **consistent** and that's far more important than realism.

For example before #15072 thirty rounds of `223` weighed 0.1kg, after it a STANAG magazine containing the same amount weighs 0.7kg.

> we're talking in terms of grams-per-capacity

I'd say this is a significant increase. Four loaded STANAG magazines is now just short of 3kg. Four loaded Glock magazines are less than 1kg.

> and if it has no real gameplay purpose

Before #15072 ammo weight wasn't nearly as much of a consideration but after it very much is.

> It DOES however have a realism effect

Incidentally how far are the above values from reality?

> there's no reason not to use real-world weights for the magazines whenever possible.

Sure, and I suspect if we used realistic values throughout then balance wouldn't be problematic. Execpt this isn't liable to happen with multiple authors and indeed there's considerable evidence in the JSON's that this is the case. For example the weight for `223` was out by a factor of 3 both from a balance and realism perspective and along with many other values appears to be entirely arbitrary. The purpose of such guidelines is to avoid such errors by providing sane defaults.

> Plus it's inevitably going to draw complaints from people who are even bigger gun nerds than I am

If you can adjust a magazines weight without breaking the guidelines/balance then great. If you have  lots of such examples then perhaps the guidelines are wrong and should be changed. If you have a single item that doesn't fit with the balance in mainline then that's likely to be of limited interest and require more effort to work on then that should be mod content.

**Comment by chaosvolt** [comment]
Hmm. Granted, I'd need to actually FIND more reliable data on magazine weights first. I guess since the weight adds up, it isn't that big an issue.

**Comment by Rivet-the-Zombie** [comment]
A loaded STANAG should weigh just a bit under a pound.

**Comment by chaosvolt** [comment]
I'm not sure if that's advocating the realism or the game-ism there. o3o

**Comment by mugling** [comment]
I don't think they're necessarily mutually exclusive goals. If the above magazine weighs ~0.5kg IRL then the guideline values are reasonably close but could possibly be improved (revised downwards). This is the sort of change though that's best made when we have a number of examples that don't fit the current defaults.

**Comment by chaosvolt** [comment]
Like I said, I'd need to commit !!SCIENCE!! on this and gather as many concrete numbers as possible. Which can be done after we actually get the first steps we're working on done.

File: doc/JSON_INFO.md [hunk 2]
```diff
@@ -376,7 +376,7 @@ The syntax listed here is still valid.
 "bashing" : -5,                   // (Optional, default = 0) Bashing damage caused by using it as a melee weapon
 "to_hit" : 0,                     // (Optional, default = 0) To-hit bonus if using it as a melee weapon (whatever for?)
 "flags" : ["VARSIZE"],            // Indicates special effects, see JSON_FLAGS.md
-"magazines" : [ "stanag30"],      // Magazine types (if any) that can be used to reload this item
+"magazines" : [ "stanag30"],      // Magazine types (if any) that can be used to reload this item. The first specified is the default.
```
Comment (kevingranade) [inline doc/JSON_INFO.md]: This should move to the gun itype and if desired the tool itype.

File: doc/JSON_INFO.md [hunk 3]
```diff
@@ -396,6 +396,23 @@ The syntax listed here is still valid.
...
+"type": "MAGAZINE",   // Defines this as a MAGAZINE
+...                   // same entries as above for the generic item.
+                      // additional some magazine specific entries:
+"ammo_type": "9mm",   // What type of ammo this magazine can be loaded with
+"capacity" : 15,      // Capacity of magazine (in equivalent units to ammo charges)
+"count" : 0,          // Default amount of ammo contained by a magazine (set this for ammo belts)
+"reliability" : 8,    // How reliable this this magazine on a range of 0 to 10? (see GAME_BALANCE.md)
+"reload_time" : 100,  // How long it takes to load each unit of ammo into the magazine
+"rigid" : true,       // Volume increases proportional to contained ammo for non-rigid magazines (set this for ammo belts)
+"alternatives" : [    // Alternative magazines (if any) for use with ammo conversion mods
```
Comment (kevingranade) [inline doc/JSON_INFO.md]: This should move to the gun itype.

**Comment by kevingranade** [comment]
I like the reliability outline and the rarity balancing.
I'd also prefer to base magazine weight/volume on real numbers, but until we have real numbers to base it on, this is good.
Feel free to move the magazine compatibility entries around at your leisure, I just mentioned them as a reminder.

**Comment by chaosvolt** [comment]
Right. Now that things have calmed down a little, will be easier to move the magazine definitions to the guns.

Might try and hunt down more field manuals too, possibly the most reliable source for magazine weights for future use.

**Pull request merged**